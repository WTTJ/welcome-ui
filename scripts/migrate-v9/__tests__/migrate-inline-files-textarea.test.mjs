import { readFileSync } from 'fs'
import { dirname, resolve } from 'path'
import { fileURLToPath } from 'url'

import { describe, expect, it } from 'vitest'

import { copyDirSync, deleteDirRecursive } from '../helpers/file-utils.mjs'
import { migrateAll } from '../index.mjs'

const __dirname = dirname(fileURLToPath(import.meta.url))

describe('Inline Migration - Textarea Components', () => {
  it('migrates Textarea components with form props to Textarea with classNames', async () => {
    const fixturesDir = resolve(__dirname, './__fixtures__')
    const tempDir = resolve(__dirname, '../temp/inline-textarea-test')

    // Clean up and copy fixtures
    deleteDirRecursive(tempDir)
    copyDirSync(fixturesDir, tempDir)

    // Run migration
    await migrateAll(tempDir, { copyDir: false, interactive: false })

    // Read results
    const migratedComponent = readFileSync(resolve(tempDir, 'TextareaComponent.tsx'), 'utf8')

    // Verify with snapshots
    expect(migratedComponent).toMatchSnapshot('textarea-component.tsx')

    // Cleanup
    deleteDirRecursive(tempDir)
  })
})
