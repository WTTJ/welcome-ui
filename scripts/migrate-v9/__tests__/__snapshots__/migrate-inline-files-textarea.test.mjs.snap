// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Inline Migration - Textarea Components > migrates Textarea components with form props to Textarea with classNames > textarea-component.tsx 1`] = `
"import { Textarea } from 'welcome-ui/Textarea'

const register = (name: string, options?: object) => {
  return { name, ...options }
}

export const TextareaComponent = () => {
  return (
    <form>
      <Textarea className="mt-md" name="i18n_description_en" {...register('i18n_description_en')} />
      <Textarea placeholder="Enter description" rows={4} />
    </form>
  )
}
"
`;
